--- origsrc/midori-0.1.1/midori/wscript_build	2008-12-01 02:07:11.000000000 -0600
+++ src/midori-0.1.1/midori/wscript_build	2009-01-01 11:26:31.784375000 -0600
@@ -6,5 +6,5 @@
 obj.target = 'midori'
 obj.includes = '. ..'
 obj.find_sources_in_dirs ('.')
-obj.uselib = 'UNIQUE LIBSOUP GMODULE GIO GTK SQLITE WEBKIT LIBXML'
+obj.uselib = 'UNIQUE LIBSOUP GMODULE GTHREAD GIO GTK SQLITE WEBKIT LIBXML'
 obj.uselib_local = 'katze'
--- origsrc/midori-0.1.1/wscript	2008-12-01 02:07:11.000000000 -0600
+++ src/midori-0.1.1/wscript	2009-01-01 11:26:17.003125000 -0600
@@ -111,6 +111,7 @@
     conf.check_message_custom ('history database', 'support', sqlite)
 
     conf.check_pkg ('gmodule-2.0', destvar='GMODULE', vnum='2.8.0', mandatory=False)
+    conf.check_pkg ('gthread-2.0', destvar='GTHREAD', vnum='2.8.0', mandatory=True)
     conf.check_pkg ('gio-2.0', destvar='GIO', vnum='2.16.0', mandatory=False)
     conf.check_pkg ('gtk+-2.0', destvar='GTK', vnum='2.10.0', mandatory=True)
     conf.check_pkg ('webkit-1.0', destvar='WEBKIT', vnum='0.1', mandatory=True)
