--- origsrc/midori-0.1.7/midori/midori-app.c	2009-05-28 16:47:01.000000000 -0500
+++ src/midori-0.1.7/midori/midori-app.c	2009-06-16 14:14:54.017999600 -0500
@@ -851,7 +851,11 @@ static void
 midori_app_init_libnotify (MidoriApp* app)
 {
     gint i;
+#ifdef G_WITH_CYGWIN
+    const gchar* sonames[] = { "cygnotify-1.dll", NULL };
+#else
     const gchar* sonames[] = { "libnotify.so", "libnotify.so.1", NULL };
+#endif
 
     for (i = 0; sonames[i] != NULL && app->libnotify_module == NULL; i++ )
     {
