Unfortunately this hack breaks parallel make. :-(

--- origsrc/midori-0.1.5/extensions/wscript_build	2009-03-22 16:49:18.000000000 -0500
+++ src/midori-0.1.5/extensions/wscript_build	2009-03-30 00:50:59.875000000 -0500
@@ -26,5 +26,5 @@ for extension in extensions:
     obj.includes = '..'
     obj.source = source
     obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML'
-    obj.uselib_local = 'katze'
     obj.install_path = '${LIBDIR}/midori'
+    obj.ldflags = 'default/midori/libmidoriext.a'
--- origsrc/midori-0.1.5/midori/wscript_build	2009-03-22 16:49:18.000000000 -0500
+++ src/midori-0.1.5/midori/wscript_build	2009-03-30 00:50:54.031250000 -0500
@@ -35,3 +35,4 @@ obj.includes = '. .. ../panels'
 obj.source = 'main.c'
 obj.uselib = 'UNIQUE LIBSOUP GMODULE GTHREAD GIO GTK SQLITE WEBKIT LIBXML'
 obj.uselib_local = 'panels'
+obj.ldflags = '-Wl,--export-all-symbols,--out-implib,default/midori/libmidoriext.a'
