--- origsrc/midori-0.1.8/extensions/wscript_build	2009-07-20 16:35:01.000000000 -0500
+++ src/midori-0.1.8/extensions/wscript_build	2009-07-23 13:15:09.063628200 -0500
@@ -31,4 +31,5 @@ for extension in extensions:
     obj.includes = '..'
     obj.source = source
     obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML'
+    obj.ldflags = 'default/midori/libmidori.a'
     obj.install_path = '${LIBDIR}/midori'
--- origsrc/midori-0.1.8/midori/wscript_build	2009-07-20 16:35:01.000000000 -0500
+++ src/midori-0.1.8/midori/wscript_build	2009-07-23 13:02:56.998888900 -0500
@@ -6,7 +6,7 @@ import platform
 
 obj = bld.new_task_gen ('cc', 'staticlib')
 obj.name = 'midori-core'
-obj.target = 'midori'
+obj.target = 'midori-core'
 obj.includes = '. ..'
 obj.find_sources_in_dirs ('.', excludes=['main.c'])
 obj.add_marshal_file ('marshal.list', 'midori_cclosure_marshal')
@@ -31,3 +31,4 @@ if bld.env['WINRC']:
     obj.source += ' ../data/midori.rc'
 obj.uselib = 'UNIQUE LIBSOUP GMODULE GTHREAD GIO GTK SQLITE WEBKIT LIBXML'
 obj.uselib_local = 'panels'
+obj.linkflags = '-Wl,--export-all-symbols,--out-implib,default/midori/libmidori.a'
