--- origsrc/midori-0.5.2/midori/midori-session.c
+++ src/midori-0.5.2/midori/midori-session.c
@@ -182,6 +182,7 @@ midori_load_soup_session (gpointer setti
         "/etc/ssl/certs/ca-bundle.crt",
         "/usr/local/share/certs/ca-root-nss.crt", /* FreeBSD */
         "/var/lib/ca-certificates/ca-bundle.pem", /* openSUSE */
+        "/usr/ssl/certs/ca-bundle.crt", /* Cygwin */
         NULL
     };
     guint i;
